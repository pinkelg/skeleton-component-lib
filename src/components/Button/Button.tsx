import React, { ReactElement } from "react";

export interface ButtonProps {
  label: string;
}

const Button = (props: ButtonProps): ReactElement<HTMLButtonElement> => {
  return <button>{props.label}</button>;
};

export default Button;
