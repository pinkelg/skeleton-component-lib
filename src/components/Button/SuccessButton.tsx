import React from "react";
import Button ,{ButtonProps} from './Button'

const SuccessButton = (props: ButtonProps) => {
    return <Button {...props} label={"success"}/>
};

export default SuccessButton;