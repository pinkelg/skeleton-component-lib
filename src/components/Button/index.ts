export { default as Button } from "./Button";
export { default as SuccessButton } from "./SuccessButton";
