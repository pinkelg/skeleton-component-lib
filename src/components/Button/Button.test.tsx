import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./Button";

const defaultProps = {
  onClick: jest.fn(),
  children: "Button"
};

describe("<Button />", () => {
  it("should render", () => {
    render(<Button {...defaultProps} />);

    expect(screen.getByTestId("Button-Button")).toMatchSnapshot();
  });

  it("should call onClick", () => {
    render(<Button {...defaultProps} />);
    fireEvent.click(screen.getByTestId("Button-Button"));

    expect(defaultProps.onClick).toHaveBeenCalledTimes(1);
  });

  it("should be disableable", () => {
    render(<Button {...defaultProps} disabled />);

    expect(screen.getByTestId("Button-Button")).toHaveProperty("disabled");
  });

  it("should render children", () => {
    render(<Button {...defaultProps} disabled />);

    expect(screen.getByText(/Button/i)).toBeInTheDocument();
  });

  it("should allow custom className", () => {
    const props = {
      ...defaultProps,
      className: "Custom"
    };
    render(<Button {...props} />);

    expect(screen.getByTestId("Button-Button")).toHaveProperty("className");
  });
});
