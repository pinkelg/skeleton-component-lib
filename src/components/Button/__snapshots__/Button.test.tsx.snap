// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render 1`] = `
<button
  class="button"
  data-testid="Button-Button"
  type="button"
>
  Button
</button>
`;
